### PR TITLE
Fix body_json matching and implement Partial json matching

### PR DESCRIPTION
**Pull request by 95th**
Fix body_json matching and implement Partial json matching.

Closes https://github.com/LukeMathWalker/wiremock-rs/issues/79

### CODE REVIEW

**Comment by LukeMathWalker** [comment]
Hey!
Thanks for the PR - the fix looks alright, but I'd prefer to avoid introducing breaking changes for this issue. Could you rework the implementation to backwards-compatible?

**Comment by 95th** [comment]
@LukeMathWalker I changed it to use an enum for body and kept the member functions.

**Comment by LukeMathWalker** [comment]
Looking good!
The partial JSON matcher is its own feature and it should be its own matcher - having a `partial` method on something named `BodyExactMatcher` is not great.

**Comment by 95th** [comment]
yes. I thought so. let me move it out.

**Comment by 95th** [comment]
How about in `BodyContainsMatcher`? Or a separate type?

**Comment by 95th** [comment]
Too late. already done.

File: src/matchers.rs [hunk 1]
```diff
@@ -626,6 +654,78 @@ impl Match for BodyContainsMatcher {
...
+///         .status();
+///
+///     // Assert
+///     assert_eq!(status, 200);
+/// }
+/// ```
+pub struct BodyPartialJsonMatcher(Value);
+
+impl BodyPartialJsonMatcher {
+    /// Specify the part of the body that should be matched as a string.
```
Comment (LukeMathWalker) [inline src/matchers.rs:697]:
```suggestion
    /// Specify the part of the body that should be matched as a JSON value.
```

**Review by LukeMathWalker — APPROVED** [review]

**Comment by LukeMathWalker** [comment]
Thanks for the PR!

**Pull request merged**